Source/misdat.cpp

```diff
@@ -10,182 +10,182 @@ namespace devilution {
-	{  MIS_RUNEFIRE,      &AddFireRune,               &MI_Rune,           true,      1, MISR_NONE,      MFILE_RUNE,      -1,          -1          },
-	{  MIS_RUNELIGHT,     &AddLightningRune,          &MI_Rune,           true,      1, MISR_NONE,      MFILE_RUNE,      -1,          -1          },
-	{  MIS_RUNENOVA,      &AddGreatLightningRune,     &MI_Rune,           true,      1, MISR_NONE,      MFILE_RUNE,      -1,          -1          },
-	{  MIS_RUNEIMMOLAT,   &AddImmolationRune,         &MI_Rune,           true,      1, MISR_NONE,      MFILE_RUNE,      -1,          -1          },
-	{  MIS_RUNESTONE,     &AddStoneRune,              &MI_Rune,           true,      1, MISR_NONE,      MFILE_RUNE,      -1,          -1          },
-	{  MIS_HIVEEXP,       &AddRuneExplosion,          &MI_HiveExplode,    true,      1, MISR_FIRE,      MFILE_BIGEXP,    LS_NESTXPLD, LS_NESTXPLD },
-	{  MIS_HORKDMN,       &AddHorkSpawn,              &MI_HorkSpawn,      true,      2, MISR_NONE,      MFILE_NONE,      -1,          -1          },
-	{  MIS_JESTER,        &AddJester,                 &MI_Dummy,          false,     2, MISR_NONE,      MFILE_NONE,      -1,          -1          },
-	{  MIS_HIVEEXP2,      &AddHiveExplosion,          &MI_Dummy,          false,     2, MISR_NONE,      MFILE_NONE,      -1,          -1          },
-	{  MIS_LICH,          &AddFlare,                  &MI_Firebolt,       true,      1, MISR_MAGIC,     MFILE_LICH,      -1,          -1          },
-	{  MIS_PSYCHORB,      &AddFlare,                  &MI_Firebolt,       true,      1, MISR_MAGIC,     MFILE_BONEDEMON, -1,          -1          },
-	{  MIS_NECROMORB,     &AddFlare,                  &MI_Firebolt,       true,      1, MISR_MAGIC,     MFILE_NECROMORB, -1,          -1          },
-	{  MIS_ARCHLICH,      &AddFlare,                  &MI_Firebolt,       true,      1, MISR_MAGIC,     MFILE_ARCHLICH,  -1,          -1          },
-	{  MIS_BONEDEMON,     &AddFlare,                  &MI_Firebolt,       true,      1, MISR_MAGIC,     MFILE_BONEDEMON, -1,          -1          },
-	{  MIS_EXYEL2,        &AddMisexp,                 &MI_Misexp,         true,      2, MISR_NONE,      MFILE_EXYEL2,    LS_FIRIMP2,  -1          },
-	{  MIS_EXRED3,        &AddMisexp,                 &MI_Misexp,         true,      2, MISR_NONE,      MFILE_EXRED3,    LS_FIRIMP2,  -1          },
-	{  MIS_EXBL2,         &AddMisexp,                 &MI_Misexp,         true,      2, MISR_NONE,      MFILE_EXBL2,     LS_FIRIMP2,  -1          },
-	{  MIS_EXBL3,         &AddMisexp,                 &MI_Misexp,         true,      2, MISR_NONE,      MFILE_EXBL3,     LS_FIRIMP2,  -1          },
-	{  MIS_EXORA1,        &AddMisexp,                 &MI_Misexp,         true,      2, MISR_NONE,      MFILE_EXORA1,    LS_FIRIMP2,  -1          },
+	// mAddProc,                   mProc,              mName,             mDraw, mType, mResist,        mFileNum,        miSFX,       mlSFX;

```

Comment:
Any specific reason to swap around the fields?

Reply:
To reduce the size of the struct by reducing the required padding

A struct like this takes up 16 bytes

```
uint8_t a;
uint64_t b;
```

A struct like this takes up a lot less:

```
uint64_t b;
uint8_t b;
```

Reply:
Oh yeah, I dealt a lot with that in the early stage of this project that's basically how I read structs at this point :D